ledger/participant-local-store/src/main/scala/com/daml/platform/localstore/InMemoryIdentityProviderConfigStore.scala

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.localstore
+
+import com.daml.ledger.api.domain
+import com.daml.ledger.api.domain.{IdentityProviderConfig, IdentityProviderId}
+import com.daml.logging.LoggingContext
+import com.daml.platform.localstore.api.IdentityProviderConfigStore._
+import com.daml.platform.localstore.api.{IdentityProviderConfigStore, IdentityProviderConfigUpdate}
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.Future
+
+class InMemoryIdentityProviderConfigStore(maxIdentityProviderConfigs: Int = 10)
+    extends IdentityProviderConfigStore {
+  private val state: TrieMap[IdentityProviderId.Id, IdentityProviderConfig] =
+    TrieMap[IdentityProviderId.Id, IdentityProviderConfig]()
+
+  override def createIdentityProviderConfig(identityProviderConfig: domain.IdentityProviderConfig)(
+      implicit loggingContext: LoggingContext
+  ): Future[Result[domain.IdentityProviderConfig]] = withState {
+    for {
+      _ <- checkIssuerDoNotExists(identityProviderConfig.issuer)
+      _ <- checkIdDoNotExists(identityProviderConfig.identityProviderId)
+      _ = state.put(identityProviderConfig.identityProviderId, identityProviderConfig)
+      _ <- tooManyIdentityProviderConfigs()
```

Comment:
shouldn't we check for "too many configs" prior to attempting to change the state?

Reply:
I was following a pattern @pbatko-da used in user management for the user rights.. I suspect transaction will be rolled back

Reply:
Checking before doesn't work as another transaction can insert additional configs after the check but before this transactions commits

Reply:
@skisel-da normally a transaction will be rolled back if you throw an exception and it seems it might not work in your setup when you're working with Eithers